experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeTypes.ts

```diff
@@ -468,6 +468,12 @@ export type FlexibleNodeContent<TTypes extends AllowedTypes> = SchemaAware.Allow
 	TTypes
 >;
 
+/**
+ * Type to ensures two types overlap in some way.
+ * @alpha
+ */
+export type CheckTypesOverlap<T, TCheck> = Extract<T, TCheck> extends never ? never : T;
```

Comment:
Do we expect consumers to use this? If so, might be worth adding some `@remarks` or `@examples` explaining its use.

Reply:
I don't think so, no, as it's fairly simple and is there to enforce moving nodes between two lists must be _possible_ (and then verified at runtime).

Reply:
Should we not be exporting it then?

Reply:
Oh, it's used in type exported method signatures. I see. Given that we do have to export it, it might be worth elaborating a bit more. That way if a consumer stumbles onto it, they can get a reasonable understanding of why it's there (nit).

Reply:
done